libvast/include/vast/detail/series_builders.hpp

```diff
@@ -344,6 +339,63 @@ struct series_builder : series_builder_base {
     }
   }
 
+  template <concrete_type NewType, concrete_type CurrentType, class... Args>
+  auto change_builder(const NewType& new_value_type, auto value_to_add,
+                      auto array, const CurrentType& curr_type,
+                      type_list<Args...>) -> caf::error {
```

Comment:
The array should also likely be taken by const-ref. It also seems to always be a `std::shared_ptr<arrow::Array>` exactly, so why make that an inferred type?

Reply:
fixed in 3ed557e15972dbeba1297eb6b23724a0bab3132f